### PR TITLE
Prefault option in storage engine

### PR DESCRIPTION
**Pull request by michalbiesek**
- Add prefault option used in PMEM to slab and cuckoo configuration
~~- Add possibility to pass custom storage engine configuration for benchmarks~~
~~- Fix logging options in benchmark~~

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pelikan/8)

### CODE REVIEW

File: src/storage/cuckoo/cuckoo.h [hunk 1]
```diff
@@ -18,16 +18,18 @@
...
+/*          name                      type                default             description */
+#define CUCKOO_OPTION(ACTION)                                                                          \
+    ACTION( cuckoo_displace,          OPTION_TYPE_UINT,   CUCKOO_DISPLACE,    "# displaces allowed"   )\
+    ACTION( cuckoo_item_cas,          OPTION_TYPE_BOOL,   CUCKOO_ITEM_CAS,    "support cas in items"  )\
+    ACTION( cuckoo_item_size,         OPTION_TYPE_UINT,   CUCKOO_ITEM_SIZE,   "item size (inclusive)" )\
+    ACTION( cuckoo_nitem,             OPTION_TYPE_UINT,   CUCKOO_NITEM,       "# items allocated"     )\
+    ACTION( cuckoo_policy,            OPTION_TYPE_UINT,   CUCKOO_POLICY,      "evict policy"          )\
+    ACTION( cuckoo_max_ttl,           OPTION_TYPE_UINT,   CUCKOO_MAX_TTL,     "max ttl in seconds"    )\
+    ACTION( cuckoo_datapool,          OPTION_TYPE_STR,    CUCKOO_DATAPOOL,    "path to data pool"     )\
+    ACTION( cuckoo_prefault_datapool, OPTION_TYPE_BOOL,   CUCKOO_PREFAULT,    "prefault data pool"    )
```
Comment (pbalcer) [inline src/storage/cuckoo/cuckoo.h]: `cuckoo_datapool_prefault`
same for slab.

File: benchmarks/bench_storage.c [hunk 2]
```diff
@@ -22,15 +23,17 @@ static __thread unsigned int rseed = 1234; /* XXX: make this an option */
...
-
-#define O(b, opt) option_uint(&(b->options.benchmark.opt))
+    ACTION(entry_min_size,    OPTION_TYPE_UINT, 64,    "Min size of cache entry")\
+    ACTION(entry_max_size,    OPTION_TYPE_UINT, 64,    "Max size of cache entry")\
+    ACTION(nentries,          OPTION_TYPE_UINT, 1000,  "Max total number of cache entries" )\
+    ACTION(nops,              OPTION_TYPE_UINT, 100000,"Total number of operations")\
+    ACTION(pct_get,           OPTION_TYPE_UINT, 80,    "% of gets")\
+    ACTION(pct_put,           OPTION_TYPE_UINT, 10,    "% of puts")\
+    ACTION(pct_rem,           OPTION_TYPE_UINT, 10,    "% of removes")\
+    ACTION(storage_cfg_path,  OPTION_TYPE_STR , NULL,  "Path to storage config")
```
Comment (pbalcer) [inline benchmarks/bench_storage.c]: In the original version of this benchmark, the config file for various modules was composable, i.e., you could have made a config with options for the benchmark alongside options for the storage engine.
Is that no longer true?

Comment (michalbiesek) [reply benchmarks/bench_storage.c]: I introduce regression in cd767666539f2aca9b461bbfb28ac3cc47b1be39 - sorry about that. I will fix benchmarks in separate PR.

File: src/storage/cuckoo/cuckoo.h (hunk 1 above)
Comment (michalbiesek) [reply src/storage/cuckoo/cuckoo.h]: Done.

**Review by pbalcer — APPROVED** [review]

**Pull request merged**